internal/linker/linker.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2022, The Garble Authors.
+// See LICENSE for licensing information.
+
+package linker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+const MagicValueEnv = "GARBLE_LNK_MAGIC"
+
+type overlayFile struct {
+	Replace map[string]string
+}
+
+func compileLinker(workingDirectory string, overlay map[string]string, outputLinkPath string) error {
+	file, _ := json.Marshal(&overlayFile{Replace: overlay})
+	overlayPath := filepath.Join(workingDirectory, "overlay.json")
+	if err := os.WriteFile(overlayPath, file, os.ModePerm); err != nil {
+		return err
+	}
+
+	cmd := exec.Command("go", "build", "-overlay", overlayPath, "-o", outputLinkPath, "cmd/link")
+	cmd.Env = append(os.Environ(), "GOOS="+runtime.GOOS, "GOARCH="+runtime.GOARCH)
```

Comment:
document why we need to set GOOS and GOARCH. presumably it is because the `garble build` command may itself be cross-compiling for a different platform.